### ISSUE TITLE
Allow editing of billing and shipping addresses for customers during checkout

### ISSUE THREAD

**Issue by manuel3108** [nopSolutions/nopCommerce#6039]
nopCommerce version: 4.40.x and above

Steps to reproduce the problem: 
1. Go through the checkout and create two different addresses (one for billing & one for shipping)
2. Make sure your addresses are correctly created (i.e. by creating an order)
3. Now imagine that your part of your company moved, the billing address should be the same, but the shipping address should change
4. Now you would leave the billing address the same, and only edit the shipping address
5. You will see that you can only edit billing addresses, editing shipping addresses is apparently not possible at the moment.

**Comment by AndreiMaz** [comment]
_"editing shipping addresses is apparently not possible"_

@manuel3108 what exactly do you mean not possible? just enter a new address on the "Shipping address" step during checkout

**Comment by manuel3108** [comment]
@AndreiMaz Yeah, but during checkout you have the possibility to edit your billing address. But the possibility to edit an existing shipping address is not available.

Below is a screenshot showing a button to edit a billing addresses
![grafik](https://user-images.githubusercontent.com/30698007/146736924-ec13faca-227e-4bbf-b1c4-de81dd307a89.png)

But the button (and the functionality behind it) does not exist for shipping addresses:
![grafik](https://user-images.githubusercontent.com/30698007/146737161-5b97a46e-e499-41b4-9976-3c0ca8a42811.png)


Sure, you could just create a new address, but that's often not what you want. 
That's why I suggest adding the functionality, to also edit shipping addresses.

**Comment by glhays** [comment]
Is there order historical reasons not to edit a ship to address?


________________________________

**Comment by manuel3108** [comment]
I don't know of any reason that's why I created this issue.
At this point the order is not even created,  so I cannot imagine of any reason

**Comment by glhays** [comment]
I meant if editing an address on the fly and it was associated to a prior order.


________________________________

**Comment by manuel3108** [comment]
Thats a good point, but i guess that you could argue with the same argument why the billing address is editable.

Maybe we should change the question and ask why we have an inconsistent behaviour between the editing of billing and shipping addresses.

**Comment by glhays** [comment]
Just checked some of the major platforms they allow editing during the checkout progress, so yes why the inconsistencies here?


________________________________

**Comment by skoshelev** [comment]
Hi, @manuel3108. By default, editing the billing address is available only for unregistered users, solely for their convenience when placing multiple orders, we do not think that it is possible that the user will change the delivery address during the lifetime of the guest token. For registered users, we do not offer to edit even a billing address, since it is important for us to keep the complete history of user orders. If it is necessary to change the addresses, the customer just needs to create a new one.

Closed #6039

**Comment by manuel3108** [comment]
Ok, that's a valid point.
But we should probably still consider allowing to edit an address, even if we just clone the address in the background.

**Comment by manuel3108** [comment]
@skoshelev  During my workaround for this issue, I found out that this no longer seems to be true:

> For registered users, we do not offer to edit even a billing address, since it is important for us to keep the complete history of user orders. 

As I found out during my debugging session is that during checkout the addresses are already cloned to preserve order history. See here.

https://github.com/nopSolutions/nopCommerce/blob/a3e1426c0269225cc06bfd3cd27223596b4c0dd4/src/Libraries/Nop.Services/Orders/OrderProcessingService.cs#L505

https://github.com/nopSolutions/nopCommerce/blob/a3e1426c0269225cc06bfd3cd27223596b4c0dd4/src/Libraries/Nop.Services/Orders/OrderProcessingService.cs#L606 

I would propose that we re-open this issue, change the title to something to "allow editing of billing and shipping addresses for registered users" and treat this as a feature request.
